File: src/main/java/flustix/fluxifyed/utils/FormatUtils.java
package flustix.fluxifyed.utils;

import java.util.concurrent.TimeUnit;

public class FormatUtils {
    private static final long yearMillis = TimeUnit.DAYS.toMillis(365);
    private static final long monthMillis = TimeUnit.DAYS.toMillis(30);
    private static final long dayMillis = TimeUnit.DAYS.toMillis(1);
    private static final long hourMillis = TimeUnit.HOURS.toMillis(1);
    private static final long minuteMillis = TimeUnit.MINUTES.toMillis(1);
    private static final long secondMillis = TimeUnit.SECONDS.toMillis(1);

    public static String timeToString(long time) {
        long years = time / yearMillis;
        long months = (time % yearMillis) / monthMillis;
        long days = (time % monthMillis) / dayMillis;
        long hours = (time % dayMillis) / hourMillis;
        long minutes = (time % hourMillis) / minuteMillis;
        long seconds = (time % minuteMillis) / secondMillis;

        return (years > 0 ? years + "y " : "") + (months > 0 ? months + "mo " : "") + (days > 0 ? days + "d " : "") + (hours > 0 ? hours + "h " : "") + (minutes > 0 ? minutes + "m " : "") + (seconds > 0 ? seconds + "s " : "");
    }

    public static String formatNumber(long number) {
        if (number < 1000) {
            return String.valueOf(number);
        } else if (number < 1000000) {
            return String.format("%.1fK", number / 1000D);
        } else if (number < 1000000000) {
            return String.format("%.1fM", number / 1000000D);
        } else {
            return String.format("%.1fB", number / 1000000000D);
        }
    }
}

File: Fluxifyed/Utils/FormatUtils.cs
namespace Fluxifyed.Utils; 

public static class FormatUtils {
    private const long Second = 1000;
    private const long Minute = 60 * Second;
    private const long Hour = 60 * Minute;
    private const long Day = 24 * Hour;
    private const long Month = 30 * Day;
    private const long Year = 365 * Day;
    
    public static string FormatTime(long time, bool ms = true) {
        var years = time / Year;
        time %= Year;
        var months = time / Month;
        time %= Month;
        var days = time / Day;
        time %= Day;
        var hours = time / Hour;
        time %= Hour;
        var minutes = time / Minute;
        time %= Minute;
        var seconds = time / Second;
        time %= Second;
        var milliseconds = time;
        
        var formatted = "";
        if (years > 0) formatted += $"{years}y ";
        if (months > 0) formatted += $"{months}mo ";
        if (days > 0) formatted += $"{days}d ";
        if (hours > 0) formatted += $"{hours}h ";
        if (minutes > 0) formatted += $"{minutes}m ";
        if (seconds > 0) formatted += $"{seconds}s ";
        if (milliseconds > 0 && ms) formatted += $"{milliseconds}ms ";
        return formatted.Trim();
    }

    public static string FormatName(string name) {
        if (string.IsNullOrEmpty(name)) return "";
        
        if (name.EndsWith("s")) return $"{name}'";
        return $"{name}'s";
    }
}